### DISCUSSION TITLE
Should C/C++/C# variables be red

### DISCUSSION THREAD

**Discussion by Binaryify** [Binaryify/Material-Theme-vscode#661] [General]
Should C/C++/C# variables be red?
Some people think those should be red but other don't like red

**Comment by MalcolmMielle** [comment]
It should be consistent with other languages, so I'm voting for red :)

**Comment by tristan957** [comment]
It should follow what the Atom upstream does in my opinion, although I am not sure how much this theme currently strays from upstream.

**Comment by w-xy** [comment]
It seems to be gray in my Atom:
![捕获](https://user-images.githubusercontent.com/17586945/144635013-eed12aad-42d4-46aa-bf37-c2a11e7fe9a5.PNG)

and I'm voting for not red. It's such an uncomfortable feeling when your company's source codes have a lot of variables and all of them are red.

**Comment by MalcolmMielle** [comment]
It's not red because atom doesn't have semantic highlighting. It's a nice property of vscode (namely its lsp) that we loose by not coloring the variables.

I also think it's a lot of red but we loose more by having everything white since other elements are white.

It's possible to paint the variables to white by changing a setting in vscode though, if it's really problematic. I'll post an example soon if you want (I use a similar trick to differentiate attributes and variable colors).

Would it be possible for one-dark to provide the color of variable as a setting? The extension would have a sensible default value, and it could be changed by the user?

**Comment by Fry98** [comment]
Agreed with @MalcolmMielle. I probably wouldn't even mind having white variables in general but it should be consistent between all languages. I'm not a fan of setting a different color for C/C++/C# only.

**Comment by GenhaoLi** [comment]
All variables painted red might look kind of harsh.

Also I think function parameters, global variables and local variables should be distinct and easily recognized.

edit: why not refer to [One Dark Theme on Jetbrains marketplace](https://plugins.jetbrains.com/plugin/11938-one-dark-theme/)?

**Comment by MalcolmMielle** [comment]
I agree that everything red is kind of similar to everything white; it defeats the purpose of the semantic highlighting.
+1 of for the JetBrains marketplace version: it has red for attributes, white for variables, and yellow for argument from what I see in the screenshot.

____

In my personal settings, I use:

```
"editor.semanticTokenColorCustomizations": {
    "[One Dark Pro]": {
        "rules": {
            "variable.readonly:python": "#D19A66", 
            "property:python": "#9D527C", 
            "property": "#9D527C",
        },
    },
},
```
It changes attributes to a deeper shade of red to make them distinguishable.

**Comment by MoonChasing** [comment]
Since I used OneDark, the variable has become coral color, and I am used to this setting.

After the variable was set to white last time, the white color made me think that vscode had crashed, and its semantics could not be parsed, so it was not rendered. And my first instinct is not that the theme is deliberately set to white.

After setting it to other colors, such as white, for access to the structure, `a.mem`,  `a` is a variable, it is white, but `mem` is rendered in coral color, which is very awkward.

**Reply by MoonChasing** [reply]
Adding an option may be a good choice.

**Reply by Fry98** [reply]
Yeah, I first tried reinstalling the C++ extension and VS Code itself before it even occured to me it might be caused by the theme.

**Reply by MoonChasing** [reply]
> Yeah, I first tried reinstalling the C++ extension and VS Code itself before it even occured to me it might be caused by the theme.

Yeah，I even mentioned an issue of vscode. XD
